rockcraft_snap_patcher/patcher.sh

```diff
@@ -0,0 +1,50 @@
+#! /bin/bash
+
+# Rockcraft snap package patcher
+# Version 4
+
+set -xe
+
+# configurable vars
+src_snap="rockcraft_1.5.3.post51+gc99aa44.d20241011_amd64.snap" # snap package to modify
+rockcraft="." # Path to rockcraft repo
+craft_application="../craft-application/" # Path to craft_application repo
+craft_providers="../craft-providers/"  # Path to craft_providers repo
+craft_parts="../craft-parts/"  # Path to craft_providers repo
+
+
+# setup
+snap_rootfs="./.patcher/$(uuid)/"
+sudo mkdir -p "$snap_rootfs"
+
+# unpack rootfs
+unsquashfs -f -d "$snap_rootfs" "$src_snap"
+snap_name=$(yq '.name' "$snap_rootfs/meta/snap.yaml")
+# snap_src_version=$(yq '.version' "$snap_rootfs/meta/snap.yaml")
+snap_arch=$(yq '.architectures[0]' "$snap_rootfs/meta/snap.yaml") #TODO: support multi arch?
+
+
+# modify snap rootfs
+rm -rf "$snap_rootfs/lib/python3.10/site-packages/rockcraft"
```

Comment:
I'd prefer if we can infer the python version from the snapcraft.yaml in case it gets bumped in the future.

Reply:
fixed! We now follow the bin/python symlink